src/lib.rs

```diff
@@ -3,27 +3,264 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 extern crate alloc;
 
-mod pointer;
-pub use pointer::*;
-mod token;
-pub use token::*;
-pub mod index;
-pub use index::Index;
-
-mod error;
-pub use error::*;
+use core::{fmt, num::ParseIntError};
 
+// TODO(chanced): make these pub?
 mod assign;
-pub use assign::*;
+pub use assign::{Assign, AssignError, Assignment};
 mod delete;
-pub use delete::*;
+pub use delete::Delete;
 mod resolve;
-pub use resolve::*;
+pub use resolve::{Resolve, ResolveError, ResolveMut};
 
 pub mod prelude;
 
+// TODO(chanced): move these into lib.rs
 mod tokens;
 pub use tokens::*;
+mod pointer;
+pub use pointer::*;
+mod token;
+pub use token::*;
+pub mod index;
+pub use index::Index;
 
 #[cfg(test)]
 mod arbitrary;
+
```

Comment:
Also not sure I get the rationale for moving these out of a dedicate error module?

Reply:
it was mostly for convenience with the expectation that the surface was going to change. I wasn't sure where everything would end up but I don't think `error` needs a mod to itself. I'm guessing the 2 error types (`AssignError`, `ResolveError`) will get moved, either into `lib.rs` but if `assign`, `resolve`, and `delete` stay as mods, they make sense to live alongside their impl.

Reply:
It's not terrible to include these in the root. I do think an `error` module makes sense if you have global error types, but I guess you got rid of those. In that case yeah, defining them close to where they are produced is the best, lacking that here is an ok fallback.